src/app/pages/plugins/plugin-advanced-add/plugin-advanced-add.component.ts

```diff
@@ -0,0 +1,1232 @@
+    let ip4_addr_ctrl = this.formGroup.controls['ip4_addr'];
+    let ip6_addr_ctrl = this.formGroup.controls['ip6_addr'];
+
+    if (dhcp_ctrl.value != true && vnet_ctrl.value == true && ip4_addr_ctrl.value != undefined && ip4_addr_ctrl.value != '') {
+      this.ip4_interfaceField.required = true;
+      this.formGroup.controls['ip4_interface'].setValidators([Validators.required]);
+      this.formGroup.controls['ip4_interface'].updateValueAndValidity();
+    } else {
+      this.ip4_interfaceField.required = false;
+      this.formGroup.controls['ip4_interface'].clearValidators();
+      this.formGroup.controls['ip4_interface'].updateValueAndValidity();
+    }
+
+    if (dhcp_ctrl.value != true && vnet_ctrl.value == true && ip6_addr_ctrl.value != undefined && ip6_addr_ctrl.value != '') {
+      this.ip6_interfaceField.required = true;
+      this.formGroup.controls['ip6_interface'].setValidators([Validators.required]);
+      this.formGroup.controls['ip6_interface'].updateValueAndValidity();
+    } else {
+      this.ip6_interfaceField.required = false;
+      this.formGroup.controls['ip6_interface'].clearValidators();
+      this.formGroup.controls['ip6_interface'].updateValueAndValidity();
+    }
+  }
+
+  ngOnInit() {
+		this.aroute.params.subscribe(params => {
+			this.plugin_name = params['pk'];
+			const nameField = _.find(this.basicfieldConfig, {'name': 'uuid'});
+			nameField.value = this.plugin_name;
+    });
+    this.ip4_interfaceField = _.find(this.basicfieldConfig, {'name': 'ip4_interface'});
+    this.ip4_netmaskField = _.find(this.basicfieldConfig, {'name': 'ip4_netmask'});
+    this.ip6_interfaceField = _.find(this.basicfieldConfig, {'name': 'ip6_interface'});
+    this.ip6_prefixField = _.find(this.basicfieldConfig, {'name': 'ip6_prefix'});
+    this.vnet_default_interfaceField = _.find(this.networkfieldConfig, {'name': 'vnet_default_interface'});
+
+    // get interface options
+    this.ws.call('interfaces.query', [[["name", "rnin", "vnet0:"]]]).subscribe(
+      (res)=>{
+        for (let i in res) {
+          this.ip4_interfaceField.options.push({ label: res[i].name, value: res[i].name});
+          this.ip6_interfaceField.options.push({ label: res[i].name, value: res[i].name});
+          this.vnet_default_interfaceField.options.push({ label: res[i].name, value: res[i].name});
+        }
+      },
+      (res)=>{
+        new EntityUtils().handleWSError(this, res, this.dialogService);
+      }
+    );
+
+    this.formFileds = _.concat(this.basicfieldConfig, this.jailfieldConfig, this.networkfieldConfig, this.customConfig, this.rctlConfig);
+    this.formGroup = this.entityFormService.createFormGroup(this.formFileds);
+
+    for (const i in this.formFileds) {
+      const config = this.formFileds[i];
+      if (config.relation.length > 0) {
+        this.setRelation(config);
+      }
+    }
+
+    this.formGroup.controls['dhcp'].valueChanges.subscribe((res) => {
+			console.log(res);
+			
+      if (res) {
```

Comment:
Seems like this type of construct would be easier to read with inline ifs and maybe just overriding defaults.  For example, this replaces that whole if/else block:
```
if (res) {
  this.formGroup.controls['vnet'].setValue(true);
  this.formGroup.controls['bpf'].setValue(true);
}
_.find(this.basicfieldConfig, { 'name': 'vnet' }).required = (res ? true : false);
_.find(this.basicfieldConfig, { 'name': 'bpf' }).required = (res ? true: false);
```
Could be there is an easier/shorter way in TS/JS to convert res to a boolean, which would be even better.  Some quick searching suggests `!!res`, but that's not very readable.